extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -419,34 +400,56 @@ describe('amp-a4a', () => {
         a4aElement.setAttribute('height', 50);
         a4aElement.setAttribute('type', 'adsense');
         const a4a = new MockA4AImpl(a4aElement);
-        verifyNonAMPRender(a4a);
         const getAdUrlSpy = sandbox.spy(a4a, 'getAdUrl');
-        sandbox.stub(a4a, 'extractCreativeAndSignature').returns(
-          Promise.resolve({creative: mockResponse.arrayBuffer()}));
+        const onAmpCreativeRenderSpy = sandbox.spy(a4a, 'onAmpCreativeRender');
+        if (!isValidCreative) {
+          sandbox.stub(a4a, 'extractCreativeAndSignature').returns(
+            Promise.resolve({creative: mockResponse.arrayBuffer()}));
+        }
+        if (opt_failAmpRender) {
+          sandbox.stub(
+            a4a, 'renderAmpCreative_').returns(Promise.resolve(false));
+        }
         a4a.onLayoutMeasure();
         expect(a4a.adPromise_).to.be.instanceof(Promise);
-        return a4a.adPromise_.then(() => {
+        return a4a.adPromise_.then(promiseResult => {
```

Comment:
Perhaps not in this test, but somewhere, we should have a test for the different throw/catch cases.  I'm not sure we (I, anyway) fully understand those interactions.